### ISSUE TITLE
[PTR] Algalon issues

### ISSUE THREAD

**Issue by NickZedd** [gamefreedomgit/Frostmourne#1722]
[//]: # (REMBEMBER! Add links to things related to the bug using for example:)
[//]: # (https://www.wowhead.com/wotlk)

### Constellations

Currently, constellations move very slowly, most of the time they have no threat table, they never follow players based on threat, and when taunted, they simply move to the exact spot that the player was standing when taunting. 

Video from PTR. I get threat on them first so you can see that it only sometimes shows on the table, then I show the movement with taunts.
https://www.youtube.com/watch?v=igVJPsfqLR8

The constellations are supposed to have a threat table, making them go on holy paladins most of the time due to healing aggro. This can be seen in the linked video. Taunting them should put you first on threat and they should continue to follow you until someone else gets higher threat. They should also move a bit faster.

At this timestamp you can very clearly see them go to the Hpal in Classic, and later on you can see them following others who have aggro. You can also see that they do continue casting while moving.
https://youtu.be/ZqBxGh4b63M?t=189

**Important note!**
The way they work currently is functional, but more difficult than it was on any other server I'm aware of. That's not a bad thing in and of itself, but the reason I'm reporting this as an issue is because the random movement of the constellations prevents you from using strategies where you don't create very many black holes. This is because the constellations will randomly wander into the black holes and close them, leaving you with no way to escape Big bang. You can get around this issue by rooting or otherwise controlling the constellations. Another issue with the way it currently works is that it's very annoying to deal with as a tank, since they are so slow and don't follow you.

TL;DR - Keep it this way (Maybe on heroic only?) if you want to increase difficulty. But know that it can be annoying, and as far as I know, it didn't work like this anywhere else.

### Collapsing stars

The movement of collapsing stars is not random enough, and they almost always stay in the same general area. In the two videos linked below I let the stars live for 100 seconds before they died on their own, and as you can see most of them remained in the same spot.

In the video from classic, you can see that they roam around more randomly and stray further from their original spawn point.

https://www.youtube.com/watch?v=BibkpxtJvd4
https://www.youtube.com/watch?v=65Z9K2RITiY
https://youtu.be/ZqBxGh4b63M?t=54

### Despawning

**The boss despawns in all of the following situations:**

- **When the server restarts. Unfortunately, no other bosses were dead meaning I couldn't check whether resetting instances or changing difficulties while outside the raid would make him respawn.**

- **When switching from normal to heroic, the boss despawns permanently.**
Switching back to normal does not respawn him. The requirements for heroic were not met in the ID where I tested this, however I believe I have previously had the same thing happen when the requirements were met (all hardmodes killed on heroic).
It should also be tested whether the same thing happens when starting on heroic with the requirements being met, and then switching to normal. However, setting that up takes time and I've done it too many times recently.

- **When logging out for too long, even if you didn't pull the boss and start the timer.** 
One or two days before our raid on a Sunday, I set up an ID for my guild to try out heroic Algalon 25m. I did not pull the boss and simply logged out after opening the door (to check that the requirements were met correctly). The boss was no longer there on Sunday. 
A different ID where I pulled the boss did still have the boss, and the timer until he leaves bugged out and was stuck at 4 minutes. I speculate that this might be because a server restart will cause him to despawn as I mentioned above, and perhaps pulling him prevented that.

**Comment by Afgann** [comment]
fixed
